src/olympia/files/tests/test_models.py

```diff
@@ -1071,36 +1086,47 @@ def test_webextension_crx_version_3(self):
         """
         upload = self.get_upload('webextension_crx3.crx')
         assert upload.path.endswith('.xpi')
+        assert zipfile.is_zipfile(upload.path)
+        assert upload.hash == (
+            'sha256:'
+            '8640cdcbd5e85403b0a08f1c42b9dff362ceca6a92bf61f424c9764189c58950'
+        )
         storage.delete(upload.path)
 
     def test_webextension_crx_not_a_crx(self):
         """Test to ensure we raise an explicit exception when a .crx file isn't
         a true crx (doesn't have to be caught, showing a 500 error is fine)."""
         data = b'Cr42\x02\x00\x00\x00&\x01\x00\x00\x00\x01\x00\x00'
-        with self.assertRaises(InvalidOrUnsupportedCrx) as exc:
-            FileUpload.from_post([data], filename='test.crx', size=1234)
-        assert str(exc.exception) == 'CRX file does not start with Cr24'
+        upload = FileUpload.from_post([data], filename='test.crx', size=1234)
```

Comment:
so, with the refactoring we've lost the ability to check that the individual exceptions are thrown correctly at this level - are there tests for `write_crx_as_xpi` directly?

Reply:
Not yet but I plan to add some